lib/nodejs/lobby.js

```diff
@@ -44,6 +44,24 @@ app.get( "/", function( req, res ) {
   } );
 } );
 
+app.get( "/map-list", function( req, res ) {
+  var mapDir = "documents/ITDG/maps";
+
+  // Ensure that the map directory exists, then get the list of files in the map directory
+  mkdirp( mapDir ).then( function( err ) {
```

Comment:
A little more complicated and probably not worth the change, but you could skip `mkdirp` and go straight to `readdirAsync`. You would need to catch `ENOENT` after `readdirAsync` and return an empty result. It's a little unexpected for a GET to change the server.